deeppavlov/dataset_readers/md_yaml_dialogs_reader.py

```diff
@@ -0,0 +1,331 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, softwaredata
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import json
+import os
+import re
+import tempfile
+from collections import defaultdict
+from logging import getLogger
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import yaml
+from overrides import overrides
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.data.dataset_reader import DatasetReader
+from deeppavlov.dataset_readers.dstc2_reader import DSTC2DatasetReader
+
+
+class DomainKnowledge:
+    """the DTO-like class to store the domain knowledge from the domain yaml config."""
+
+    known_entities: List
+    known_intents: List
+    known_actions: List
+    known_slots: Dict
+    response_templates: Dict
+    session_config: Dict
+
+
+log = getLogger(__name__)
+
+
+@register('md_yaml_dialogs_reader')
+class MD_YAML_DialogsDatasetReader(DatasetReader):
+    """
+    Reads dialogs from dataset composed of stories.md, nlu.md, domain.yml .
+    """
+
+    _USER_SPEAKER_ID = 1
+    _SYSTEM_SPEAKER_ID = 2
+
+    @staticmethod
+    def _data_fname(datatype):
+        assert datatype in ('trn', 'val', 'tst'), "wrong datatype name"
+        return f"stories-{datatype}.md"
+
+    @classmethod
+    @overrides
+    def read(self, data_path: str, dialogs: bool = False) -> Dict[str, List]:
+        """
+        Parameters:
+            data_path: path to read dataset from
+            dialogs: flag which indicates whether to output list of turns or
+             list of dialogs
+
+        Returns:
+            dictionary that contains
+            ``'train'`` field with dialogs from ``'stories-trn.md'``,
+            ``'valid'`` field with dialogs from ``'stories-val.md'`` and
+            ``'test'`` field with dialogs from ``'stories-tst.md'``.
+            Each field is a list of tuples ``(x_i, y_i)``.
+        """
+        domain_fname = "domain.yml"
+        nlu_fname = "nlu.md"
+        stories_fnames = tuple(self._data_fname(dt) for dt in ('trn', 'val', 'tst'))
+        required_fnames = stories_fnames + (nlu_fname, domain_fname)
+        for reqiured_fname in required_fnames:
```

Comment:
There is a type in `reqiured_fname`, but it seems consistent.

Reply:
thx! fix is coming